index.rst

```diff
@@ -1,61 +1,181 @@
+load they should expect to receive, and will act as a harness for them to run
+tests.
 
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+The simulator should:
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+ - Provide a realistic stream of alerts, both in terms of volume and in content.
+ - Be simple to install on realistic broker hardware
+ - Be well-documented
+ - Provide tools to make it clear whether a consumer is keeping up with the
+   stream volume
 
-     Caption text.
+Implementation
+==============
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+The simulator will be provided as a set of Docker containers and a
+``docker-compose`` file for coordinating those containers. To run the simulator,
+the user spins up the docker containers, and then runs a command which starts
+the flow of data. The data comes from a directory with serialized alert data on
+disk, and is paced into the containerized Kafka broker to match the expected
+rate for Rubin. The data will be repeated indefinitely, with timestamps
+adjusted, so that long-term behavior can be observed, even if only a few minutes
+of source data is available.
 
-   Feel free to delete this instructional comment.
+Alert data will be serialized using the latest Avro alert event schema. The Avro
+schema specification used will be provided along with the containers.
+
+Docker
+------
 
-:tocdepth: 1
+These containers will be provided:
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+ - A Kafka container, which runs the broker
+ - A Zookeeper container, which the broker connects to for backend state management
+ - A monitoring container, which provides a web-based UI verify that the stream
+   is being fully handled
+ - An injector container, which hosts a command that sends data to the broker
 
-.. sectnum::
+The Kafka broker and monitoring containers will be addressable from the host
+running the containers so that the user can consume the stream and check metrics
+easily.
+
+Injector and Data Rates
+-----------------------
 
-.. TODO: Delete the note below before merging new content to the master branch.
+The injector sends data to the broker. It should consume a directory on disk,
+and will repeat the data it finds there indefinitely.
 
-.. note::
+It will attempt to send data at a realistic rate, and will have options to
+increase the rate.
+
+By default, the "realistic rate" will be to send approximately 10,000 events
+every 37 seconds. The events will be sent over a 3 second period, emitted in 189
+bursts, each with a random number of events between 10 and 200 (centered at 50).
+This emulates a plausible data rate for Rubin's 189 data-producing CCDs.
```

Comment:
Should the rate be coupled to the contents of the alert?

That is, if we have a realistic set of simulated alerts, they should probably be sent based on (simulated) visit ID, rather than simply bunched by number.

Reply:
Of course, if we have “real” alerts from DECam data, we don't want to stick with their cadence — that won't be a good approximation for Rubin.

Reply:
At risk of overcomplicating this unduly, we might also make this an option: send alerts using native timestamps in the alerts + an offset, or send them in batches of N every deltaT seconds.

Reply:
Yeah, I suspect that a bunch of controls for the pacing will be nice to have, that's a good call.